perun/postprocess/regression_analysis/tools.py

```diff
@@ -169,3 +169,19 @@ def as_plot_y_dict(plot_y):
     :return: dictionary with key 'plot_y' set to plot_y
     """
     return dict(plot_y=plot_y)
+
+
+def add_analysis_to_profile(profile, analysis):
```

Comment:
Suggestion: I approve of the refactor, however, I would rename the function to something more spot on, such as "add_model_to_profile" and "analysis" parameter to "model"